### ISSUE TITLE
Dropout layer

### ISSUE THREAD

**Issue by lars-reimann** [Safe-DS/Library#848]
### Is your feature request related to a problem?

Dropout layers are a common way to regularize a neural network.

### Desired solution

* Add a new class `safeds.ml.nn.layers.DropoutLayer`.
* Superclass: `Layer`
* Constructor parameters: 
  * `probability: float`
  * Call `_check_bounds` to ensure that `probability` is between 0 and 1 (both inclusive)
* Implementation:
  * Use PyTorch's [`Dropout`](https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html) internally
  * Pass the probality along and don't work inplace.
